src/ldp/AuthenticatedLdpHandler.ts

```diff
@@ -0,0 +1,145 @@
+import { PermissionSet } from './permissions/PermissionSet';
+import { PermissionsExtractor } from './permissions/PermissionsExtractor';
+import { RequestParser } from './http/RequestParser';
+import { ResponseWriter } from './http/ResponseWriter';
+
+/**
+ * Collection of handlers needed for {@link AuthenticatedLdpHandler} to function.
+ */
+export interface AuthenticatedLdpHandlerArgs {
+  /**
+   * Parses the incoming requests.
+   */
+  requestParser: RequestParser;
+  /**
+   * Extracts the credentials from the incoming request.
+   */
+  credentialsExtractor: CredentialsExtractor;
+  /**
+   * Extracts the required permissions from the generated Operation.
+   */
+  permissionsExtractor: PermissionsExtractor;
+  /**
+   * Verifies if the requested operation is allowed.
+   */
+  authorizer: Authorizer;
+  /**
+   * Executed the operation.
+   */
+  operationHandler: OperationHandler;
+  /**
+   * Writes out the response of the operation.
+   */
+  responseWriter: ResponseWriter;
+}
+
+/**
+ * The central manager that connects all the necessary handlers to go from an incoming request to an executed operation.
+ */
+export class AuthenticatedLdpHandler implements HttpHandler {
+  private readonly requestParser: RequestParser;
+
+  private readonly credentialsExtractor: CredentialsExtractor;
+
+  private readonly permissionsExtractor: PermissionsExtractor;
+
+  private readonly authorizer: Authorizer;
+
+  private readonly operationHandler: OperationHandler;
+
+  private readonly responseWriter: ResponseWriter;
+
+  /**
+   * Creates the handler.
+   * @param args - The handlers required. None of them are optional.
+   */
+  public constructor (args: AuthenticatedLdpHandlerArgs) {
+    Object.assign(this, args);
+  }
+
+  /**
+   * Checks if the incoming request can be handled. The check is very non-restrictive and will usually be true.
+   * It is based on whether the incoming request can be parsed to an operation.
+   * @param input - Incoming request and response. Only the request will be used.
+   *
+   * @returns A promise resolving if this request can be handled, otherwise rejecting with an Error.
+   */
+  public async canHandle (input: { request: HttpRequest; response: HttpResponse }): Promise<void> {
+    return this.requestParser.canHandle(input.request);
+  }
+
+  /**
+   * Handles the incoming request and writes out the response.
+   * This includes the following steps:
+   *  - Parsing the request to an Operation.
+   *  - Extracting credentials from the request.
+   *  - Extracting the required permissions.
+   *  - Validating if this operation is allowed.
+   *  - Executing the operation.
+   *  - Writing out the response.
+   * @param input - The incoming request and response object to write to.
+   *
+   * @returns A promise resolving when the handling is finished.
+   */
+  public async handle (input: { request: HttpRequest; response: HttpResponse }): Promise<void> {
+    let err: Error = null;
```

Comment:
TS conventions recommend avoiding `null` whenever possible, and instead using `undefined`.
I think this would work well here, and would avoid the need for an initializer.

Also, could you enable `strict` mode in TS (perhaps in a separate PR)? This would require this type to become `Error | undefined`.

Reply:
It should already be in strict mode (I have `"alwaysStrict": true,` in the config, that's the only thing that's needed right?). Typescript doesn't mention that the type should change when I remove the `= null` though.